.github/workflows/update-docker-requirements.yml

```diff
@@ -42,10 +43,22 @@ jobs:
           NEEDS_UPDATE=$(git diff --quiet && echo 'false' || echo 'true')
           echo "needs-update=${NEEDS_UPDATE}" | tee --append $GITHUB_OUTPUT
 
+      - name: Get last merged PR info
+        id: pr_info
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+        run: |
+          LAST_MERGED_PR=$(gh pr list --repo $GITHUB_REPOSITORY --state closed --json number --jq '.[0].number')
+          echo "Last merged PR number: $LAST_MERGED_PR"
```

Comment:
This logic is flawed unfortunately. Just because we have the last merged PR, it doesn't have to be merged into `main`. Imagine merging a PR into a feature branch.

Instead, we need to query the API for the [PR associated with the commit](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-pull-requests-associated-with-a-commit) we are currently running the workflow on. It should be something like

```bash
RESPONSE=$(gh api \
  -H 'Accept: application/vnd.github+json' \
  -H 'X-GitHub-Api-Version: 2022-11-28' \
  /repos/{owner}/{repo}/commits/${{ github.sha }}/pulls)
if [ $? -ne 0 ]; then
  echo 'has-pr=false' | tee --append $GITHUB_OUTPUT
  exit 0
fi

echo 'has-pr=true' | tee --append $GITHUB_OUTPUT
PR_NUMBER=$(jq '.[0].number' <<< "${RESPONSE}")
```

We have to adjust the body of the PR that we open based on `${{ ${{ steps.pr-info.outputs.has-pr }} }}`.

Reply:
> Thoughts?

Your suggestion makes sense, Thanks. 
I'm on it to refine it the same way.